pkg/migrations/00020_add_latest_envelopes.up.sql

```diff
@@ -0,0 +1,37 @@
+CREATE TABLE gateway_envelopes_latest (
+    originator_node_id int PRIMARY KEY,
+    originator_sequence_id bigint NOT NULL,
+    gateway_time timestamp NOT NULL
+);
+
+CREATE OR REPLACE FUNCTION update_latest_envelope() RETURNS trigger AS $$
+BEGIN
+    INSERT INTO gateway_envelopes_latest
+    VALUES (NEW.originator_node_id, NEW.originator_sequence_id, NEW.gateway_time)
+    ON CONFLICT (originator_node_id)
+        DO UPDATE
+        SET originator_sequence_id = EXCLUDED.originator_sequence_id,
+            gateway_time = EXCLUDED.gateway_time;
```

Comment:
These can move backward. Probably want `greatest(OLD.gateway_time, EXCLUDED.gateway_time)` to enforce that they only move forward.